joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/TypeHelper.scala

```diff
@@ -0,0 +1,117 @@
+      case BabelAst.TSUndefinedKeyword  => Defines.ANY.label
+      case BabelAst.TSUnknownKeyword    => Defines.ANY.label
+      case BabelAst.TSVoidKeyword       => Defines.ANY.label
+      case BabelAst.TSThisType          => dynamicInstanceTypeStack.headOption.getOrElse(Defines.ANY.label)
+      case BabelAst.TSFunctionType      => Defines.ANY.label
+      case BabelAst.TSConstructorType   => Defines.ANY.label
+      case BabelAst.TSTypeReference     => code(tsType.json)
+      case BabelAst.TSTypePredicate     => Defines.ANY.label
+      case BabelAst.TSTypeQuery         => Defines.ANY.label
+      case BabelAst.TSTypeLiteral       => Defines.ANY.label
+      case BabelAst.TSArrayType         => code(tsType.json)
+      case BabelAst.TSTupleType         => Defines.ANY.label
+      case BabelAst.TSOptionalType      => typeForTypeAnnotation(createBabelNodeInfo(tsType.json(TYPE_ANNOTATION_KEY)))
+      case BabelAst.TSRestType          => typeForTypeAnnotation(createBabelNodeInfo(tsType.json(TYPE_ANNOTATION_KEY)))
+      case BabelAst.TSUnionType         => Defines.ANY.label
+      case BabelAst.TSIntersectionType  => Defines.ANY.label
+      case BabelAst.TSConditionalType   => Defines.ANY.label
+      case BabelAst.TSInferType         => Defines.ANY.label
+      case BabelAst.TSParenthesizedType => typeForTypeAnnotation(createBabelNodeInfo(tsType.json(TYPE_ANNOTATION_KEY)))
+      case BabelAst.TSTypeOperator      => Defines.ANY.label
+      case BabelAst.TSIndexedAccessType => Defines.ANY.label
+      case BabelAst.TSMappedType        => Defines.ANY.label
+      case BabelAst.TSLiteralType       => Defines.ANY.label
+      case BabelAst.TSExpressionWithTypeArguments => Defines.ANY.label
+      case BabelAst.TSImportType                  => Defines.ANY.label
+      case _ =>
+        notHandledYet(tsType, TYPE_NOT_HANDLED_TEXT)
+        Defines.ANY.label
+    }
+  }
+
+  private def typeForTypeAnnotation(typeAnnotation: BabelNodeInfo): String = {
+    typeAnnotation.node match {
+      case BabelAst.TypeAnnotation   => typeForFlowType(createBabelNodeInfo(typeAnnotation.json(TYPE_ANNOTATION_KEY)))
+      case BabelAst.TSTypeAnnotation => typeForTsType(createBabelNodeInfo(typeAnnotation.json(TYPE_ANNOTATION_KEY)))
+      case _: BabelAst.FlowType      => typeForFlowType(createBabelNodeInfo(typeAnnotation.json))
+      case _: BabelAst.TSType        => typeForTsType(createBabelNodeInfo(typeAnnotation.json))
+      case _ =>
+        notHandledYet(typeAnnotation, TYPE_NOT_HANDLED_TEXT)
+        Defines.ANY.label
+    }
+  }
+
+  protected def typeFor(node: BabelNodeInfo): String = {
+    if (hasKey(node.json, TYPE_ANNOTATION_KEY)) {
+      val tpe = typeForTypeAnnotation(createBabelNodeInfo(node.json(TYPE_ANNOTATION_KEY)))
+      registerType(tpe, tpe)
+      tpe
+    } else if (hasKey(node.json, RETURN_TYPE_KEY)) {
+      val tpe = typeForTypeAnnotation(createBabelNodeInfo(node.json(RETURN_TYPE_KEY)))
+      registerType(tpe, tpe)
```

Comment:
this whole fn looks _uneven_ to quote scala master @mpollmeier

Reply:
lol, i didn't say this to wrt this method, so saying "quote" is a lie :)
You could remove some duplication here though, e.g. like this:

```scala
    Seq(TYPE_ANNOTATION_KEY, RETURN_TYPE_KEY).find(hasKey(node.json, _)) match {
      case Some(key) =>
        val tpe = typeForTypeAnnotation(createBabelNodeInfo(node.json(key)))
        registerType(tpe, tpe)
        tpe
      case None =>
        Defines.ANY.label
    }
```

alternatively

```scala
    Seq(TYPE_ANNOTATION_KEY, RETURN_TYPE_KEY)
      .find(hasKey(node.json, _))
      .map { key =>
        val tpe = typeForTypeAnnotation(createBabelNodeInfo(node.json(key)))
        registerType(tpe, tpe)
        tpe
      }
      .getOrElse(Defines.ANY.label)
```

Reply:
Ugh, bad choice of words on my part, I only wanted to say that I quoted the word _uneven_ not that you called the implementation _uneven_

Reply:
note that my suggestions are even less _even_, but they don't contain unnecessary duplication

@max-leuthaeuser sorry that we're using this PR as an exercise to improve code quality, please don't take it personal :)